mffpy/writer.py

```diff
@@ -84,21 +84,21 @@ def addxml(self, xmltype, filename=None, **kwargs):
     def addbin(self, binfile: BinWriter, filename=None):
         """Add the .bin file to the collection
 
-        Currently we only allow to add one such files, b/c .mff can only have
-        one `epochs` file.  For this we added the flag `self._bin_file_added`.
-
         **Parameters**
 
         *binfile*: `class BinWriter` to be added to the collection
-        *filename*: (defaults to `binfile.default_filename`) filename of the
-            bin file.  It's not recommended to change this default value.
+
+        *filename*: (defaults to `binfile.default_filename_fmt %
+            self.num_bin_files`) filename of the bin file.  It's not
+            recommended to change this default value.
         """
-        assert not self._bin_file_added
-        self.files[filename or binfile.default_filename] = (
-            binfile, type(binfile))
-        self.addxml('dataInfo', **binfile.get_info_kwargs())
+        self.num_bin_files += 1
+        binname = filename or \
+            (binfile.default_filename_fmt % self.num_bin_files)
```

Comment:
I would recommend determining the filename of the binary and info files from the data type. As far as I can tell, `signal1.bin` is always EEG and `signal2.bin` is always PNSData. If someone tries to add a binary file with PNS data first, it will get labelled as `signal1.bin`, which may lead to issues down the road.

Reply:
I think the file type is specified in info1.xml or info2.xml.  Suppose the information were implicit in the order of numbers; it would mean that info1.xml and info2.xml contain redundant information.   I think if the reader is written well enough, it shouldn't matter.  I agree however to add a warning that it is customary to write certain files in certain orders.